services/app/src/components/prompt/WindowPrompt.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+function promptInput() {
+  if (
+    localStorage.getItem('user') === '' ||
+    localStorage.getItem('user') === null
+  ) {
+    const person = prompt('Please enter your name');
+    if (person != null) {
+      window.localStorage.setItem('user', person);
+    }
+  }
+}
+
+function WindowPrompt() {
+  return (
+    <div className="windowPrompt">
+      <script src={promptInput()} />
```

Comment:
`promptInput()` doesn't return anything and your script tag is left empty and without source which does nothing. The main action happens in js trying to figure out what the value for it's source should be.

You could return `null` from this component and call `promptInput()` from a `useEffect`:
```
function WindowPrompt() {
  useEffect(() => promptInput(), []);
  return null;
}
```

Reply:
On second though your way should actually block javascript from executing, where as mine goes through a render step and then blocks on `useEffect`.

Reply:
I think we should have some sort of peer-programming done here, it would be a great help to understand the problem and it fix